crates/state/src/wl_storage.rs

```diff
@@ -572,13 +573,27 @@ macro_rules! impl_storage_traits {
 impl_storage_traits!(WlStorage<D, H>);
 impl_storage_traits!(TempWlStorage<'_, D, H>);
 
+impl<D, H> WithConversionState for WlStorage<D, H>
```

Comment:
Does this really need a separate trait?

Reply:
I think so - it is used to replace the `state` dep in `shielded_token` with `storage`, which restricts what the crate can access. The idea is to use a similar approach (with mocking when needed) to remove inter-sub-system dependencies